src/core/ColorUtil.cc

```diff
@@ -100,4 +103,194 @@ std::optional<Color4f> parse_hex_color(const std::string& hex) {
   return rgba;
 }
 
-} // namespace
\ No newline at end of file
+std::optional<Color4f> parse_web_color(const std::string& col) {
+  std::string colorname = boost::algorithm::to_lower_copy(col);
+  if (webcolors.find(colorname) != webcolors.end()) {
+    return webcolors.at(colorname);
+  } 
+  return {};
+}
+
+std::optional<Color4f> parse_color(const std::string& col) {
+  const auto webcolor = parse_web_color(col);
+  if (webcolor) {
+    return webcolor;
+  }
+
+  const auto hexcolor = parse_hex_color(col);
+  if (hexcolor) {
+    return hexcolor;
+  }
```

Comment:
Nit: Since hexcolor is cheap to test (as it has specific length requirements and needs to start with a `#`, perhaps check it first.

Reply:
The logic to test this is already in `parse_hex_color`. It accepts four different string lengths, and testing is necessary to choose how it's parsed. I don't see any value in duplicating that logic here.